lib/split/trial.rb

```diff
@@ -96,20 +95,23 @@ def choose!(context = nil)
       alternative
     end
 
-    private
-
-    def delete_time_of_assignment_key
-      @user.delete("#{@experiment.name}:time_of_assignment")
-    end
-
     def within_conversion_time_frame?
-      window_of_time_for_conversion_in_minutes = Split.configuration.experiments.dig(@experiment.name, "window_of_time_for_conversion_in_minutes")
+      if @within_conversion_time_frame
+        @within_conversion_time_frame
+      else
+        window_of_time_for_conversion_in_minutes = Split.configuration.experiments.dig(@experiment.name, "window_of_time_for_conversion_in_minutes")
 
-      return true if window_of_time_for_conversion_in_minutes.nil?
+        return true if window_of_time_for_conversion_in_minutes.nil?
 
-      time_of_assignment = Time.parse(@user["#{@experiment.name}:time_of_assignment"])
+        time_of_assignment = Time.parse(@user["#{@experiment.name}:time_of_assignment"])
+        @within_conversion_time_frame = (Time.now - time_of_assignment)/60 <= window_of_time_for_conversion_in_minutes
+      end
```

Comment:
as discussed we can probably re-adjust this later to encapsulate the else condition in a private function and have the public return @within_conversion_time_frame || private_meth. This also resolves the handling of when window_of_time_for_conversion_in_minutes is nil and returns true ~ which we would want to potentially save as a minor optimization.